### ISSUE TITLE
Adding UI that prevents the project creation failure error from happening

### ISSUE THREAD

**Issue by jayoung-lee** [flutter/flutter-intellij#2731]
In our recent user study, the project creation failure due to upper case letter (#1058) happened again. 

In addition to fixing this issue, I'd like to suggest adding a UI that prevents this from happening, **before** hitting 'Finish' and seeing the 'Cannot Save Setting' warning message. 

One option would be showing an error message as the project name gets typed in, similar to password strength verification. Another option would be applying a helpful constraint, meaning that the text field only accepts lowercase letter and underscore.

**Comment by pq** [comment]
See also #1058; net-net: at the time this was difficult in IDEA for platform reasons.  Android Studio might be a different issue though?

**Comment by pq** [comment]
Also related: #1095.

**Comment by stevemessick** [comment]
Yes, the Android Studio version does character-by-character validation. The whole wizard framework is entirely different from the one in IntelliJ.
